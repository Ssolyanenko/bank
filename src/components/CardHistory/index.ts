export * from './History';
export * from './Pagination';
export * from './CardHistory';
