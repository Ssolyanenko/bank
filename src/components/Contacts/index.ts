export * from './Contacts';
export * from './ContactsFooter';
export * from './ContactsInformation';
