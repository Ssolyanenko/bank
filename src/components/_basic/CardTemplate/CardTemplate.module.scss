@import 'src/styles/variables.module.scss';

.cardWrappingArea {
  width: 320px;
  height: 200px;
  perspective: 600px;

  &.radioButtonPointWrapper {
    margin-left: -30px;
    margin-right: -35px;
    margin-top: 10px;
  }

  .cardWrapper {
    position: relative;
    cursor: pointer;
    transform-style: preserve-3d;
    transform-origin: center right;
    transition: transform 1s;

    .card {
      position: absolute;
      width: 320px;
      height: 200px;
      display: flex;
      flex-direction: column;
      border-radius: 12px;
      padding: 16px 24px;
      font-size: 1.25rem;
      background-color: transparent;
      backface-visibility: hidden;

      &.radioButtonPointCard {
        transform: scale(0.76);
      }

      &.notSelectedCard {
        opacity: 0.5;
      }

      .nameWrapper {
        display: flex;
        flex-direction: row;
        align-items: center;
        margin-bottom: 11px;

        .name {
          width: 100%;
          height: 37px;
          display: flex;
          align-items: center;
          font-size: 1.25rem;
          font-weight: $bold;
          color: $white_1;
          margin: 0;
        }

        .logoIcon {
          width: 59px;
          height: 29px;
        }
      }

      .creditNumber {
        height: 20px;
        width: 212px;
        margin-bottom: 8px;
        display: flex;
        justify-content: flex-start;
        align-items: center;
        gap: 16px;

        .dotNumbers {
          height: 28px;
          width: 46px;
          display: flex;
          align-items: center;
          font-size: 0.875rem;
          color: $white_1;
        }
      }

      .cardDate {
        color: $white_1;
        font-size: 0.875rem;
      }

      .totalSumWrapper {
        display: flex;
        justify-content: space-between;
        margin-top: auto;

        .totalSum {
          height: 32px;
          color: $white_1;
          margin: 0;
        }
      }
    }

    .cardBack {
      transform: rotateY(180deg);

      .stripe {
        width: 100%;
        height: 46px;
        background-color: $transparent_15;
        position: absolute;
        top: 24px;
        left: 0;
      }

      .cvv {
        height: 36px;
        width: 75px;
        background-color: $grayLight_1;
        position: absolute;
        top: 148px;
        left: 229px;
        border-radius: 4px;
        font-size: 0.875rem;
        padding: 8px 16px;
        user-select: none;

        .showPassword {
          display: flex;
          align-items: center;
          justify-content: space-between;
          width: 100%;
          line-height: 1.25;

          .stars {
            margin-top: 3px;
          }

          .icon {
            display: block;
            margin-top: 4px;
            transform: scaleX(-1);
          }

          .starsActive {
            margin-top: 0;
          }
        }
      }
    }
  }

  .cardWrapper.isFlipped {
    transform: translateX(-100%) rotateY(-180deg);
  }
}

.debitSmart {
  background: url('~/public/images/debitSmartCardBackground.png') center / cover no-repeat;
  background-origin: border-box;
}

.debitTravel {
  background: url('~/public/images/debitTravelCardBackGround.png') center / cover no-repeat;
  background-origin: border-box;
}

.billable {
  background: url('~/public/images/billableCardBackGround.png') center / cover no-repeat;
  background-origin: border-box;
}

.billablePremium {
  background: url('~/public/images/billablePremiumCardBackground.png') center / cover no-repeat;
  background-origin: border-box;
}
