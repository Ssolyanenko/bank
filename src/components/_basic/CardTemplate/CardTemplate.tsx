import { FC, ReactElement, SyntheticEvent, useState, useEffect } from 'react';

import { AppLogoIcon, PasswordIconInvisible, PasswordIconVisible } from 'assets';
import { setLogoIcon } from 'helpers/setLogoIcon';
import { setLogoIconColor } from 'helpers/setLogoIconColor';
import { CardTemplate as CardTemplateProps } from 'interfaces/cardTemplate';
import { setCardBackground, getLastFourNumbers, setCardName } from 'helpers';
import { TIMER_CVV } from 'constants/numbers';
import { SEE_CARD_INFO } from 'constants/text';
import { CardTemplateDots } from './CardTemplateDots';
import classes from './CardTemplate.module.scss';

export const CardTemplate: FC<CardTemplateProps> = ({
  cardProductName,
  formattedAmount = '',
  currency = '',
  cardDate,
  paymentSystem,
  cardNumber = '1029384756483902',
  cvv = '***',
  isRadioButtonPoint = false,
  isSelectedCard = false,
}): ReactElement => {
  const [isBackSideShow, setIsBackSideShow] = useState(false);
  const [isCvvShow, setIsCvvShow] = useState(false);

  const onShowCvv = (event: SyntheticEvent): void => {
    event.stopPropagation();
    setIsCvvShow(!isCvvShow);
  };

  useEffect((): (() => void) => {
    const timer = setTimeout((): void => {
      setIsCvvShow(false);
    }, TIMER_CVV);

    return (): void => {
      clearTimeout(timer);
    };
  });

  return (
    <div className={`${classes.cardWrappingArea} ${isRadioButtonPoint ? classes.radioButtonPointWrapper : ''}`}>
      <div
        role="presentation"
        onClick={(): void => {
          if (!isRadioButtonPoint) {
            setIsBackSideShow(!isBackSideShow);
          }
        }}
        className={`${classes.cardWrapper} ${isBackSideShow ? classes.isFlipped : ''}`}
      >
        <main
          className={` ${classes.card} ${classes[setCardBackground(cardProductName)]} ${
            isRadioButtonPoint ? classes.radioButtonPointCard : ''
          } ${!isSelectedCard && isRadioButtonPoint ? classes.notSelectedCard : ''}`}
        >
          <div className={classes.nameWrapper}>
            <h3 className={classes.name}>{setCardName(cardProductName)}</h3>
            <AppLogoIcon color={setLogoIconColor(setCardBackground(cardProductName))} className={classes.logoIcon} />
          </div>
          <div className={classes.creditNumber}>
            <CardTemplateDots />
            <CardTemplateDots />
            <CardTemplateDots />
            <div className={classes.dotNumbers}>{getLastFourNumbers(cardNumber)}</div>
          </div>
          <div className={classes.cardDate}>{cardDate}</div>
          <div className={classes.totalSumWrapper}>
            <p className={classes.totalSum}>
              {formattedAmount === SEE_CARD_INFO ? formattedAmount : `${formattedAmount} ${currency}`}
            </p>
            {setLogoIcon(paymentSystem)}
          </div>
        </main>
        <div className={` ${classes.card} ${classes[setCardBackground(cardProductName)]} ${classes.cardBack}`}>
          <div className={classes.stripe} />
          <div className={classes.cvv}>
            <div className={classes.showPassword}>
              {!isCvvShow ? (
                <>
                  <span className={classes.stars}> *** </span>
                  <span className={classes.icon}>
                    <PasswordIconInvisible onClick={onShowCvv} />
                  </span>
                </>
              ) : (
                <>
                  <span className={classes.starsActive}>{cvv}</span>
                  <span className={classes.icon}>
                    <PasswordIconVisible onClick={onShowCvv} />
                  </span>
                </>
              )}
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};
