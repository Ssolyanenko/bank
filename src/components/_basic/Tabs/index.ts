export * from './Tabs';
export * from './TabTitle';
