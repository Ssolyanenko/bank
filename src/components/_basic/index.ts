export * from './PrimaryButton';
export * from './Button';
export * from './Input';
export * from './InputPassword';
export * from './ModalWrapper';
export * from './ModalConfirmation';
export * from './PhoneCodeTimer';
export * from './ModalBox';
export * from './CloseButton';
export * from './BackButton';
export * from './StepperForm';
export * from './StepperLabel';
export * from './ChatButton';
export * from './BanTimer';
export * from './CardTemplate';
export * from './SecondaryButton';
export * from './generic';
export * from './SelectStyled';
export * from './TextFieldStyled';
export * from './InputStyled';
export * from './CardTemplate';
export * from './Modal';
export * from './SecureWrapper';
export * from './InputRange';
export * from './DropDown';
export * from './RadioButtonsGroup';
export * from './FormError';
export * from './Checkbox';
export * from './Switcher';
export * from './EditButton';
export * from './CardBlock';
export * from './Tabs';
export * from './ModalSuccess';
export * from './DropDownWithCheckbox';
export * from './SelectFormikTemplate';
export * from './Loader';
export * from './DatePicker';
export * from './CustomRoutes';
