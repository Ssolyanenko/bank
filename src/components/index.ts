export * from './LandingPageComponents';
export * from './Forms';
export * from './ArrowsBlock';
export * from './RootModal';
export * from './Slider';
export * from './Map';
export * from './Contacts';
export * from './MainPageComponents';
export * from './ExchangeRates';
export * from './CardProducts';
export * from './CardHistory';
export * from './SetPinSuccessModal';
export * from './MyCardsApplications';
export * from './MyLoans';
export * from './MyCardsApplications';
export * from './InformationBlock';
export * from './StatusTag';
export * from './DropDownWithSearchBranches';
export * from './FormInputBox';
export * from './MyCards';
export * from './ActivateBlockCardModal';
export * from './UserAccount';
export * from './SetPinCodeFormModal';
export * from './CardDetails';
export * from './CardStatement';
export * from './CardStatementButtons';
export * from './Accordion';
export * from './ReceiptButtons';
export * from './TemporaryComponent/TemporaryComponent';
export * from './TransferModal';
export * from './ModalChoosePaymentCard';
export * from './CardInfo';
export * from './ShareReceiptModal/ShareReceiptModal';
export * from './SocialMedia';
export * from './ResetFormsConfirmModal';
export * from './TransactionLimit';
export * from './ChangeTransactionLimit';
export * from './ActivateBlockCard';
export * from './CardManage';
export * from './ChangePINCode';
export * from './ResponseModal';
export * from './TransactionLimit';
