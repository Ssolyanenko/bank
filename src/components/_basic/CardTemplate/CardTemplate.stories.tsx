import React, { ReactElement } from 'react';
import { ComponentMeta } from '@storybook/react';

import { CardTemplate as CardTemplateProps } from 'interfaces/cardTemplate';
import { CardTemplate } from 'components/_basic/CardTemplate';
import { PaymentSystem } from 'constants/paymentSystem';
import { convertUnixTimestampToDate } from 'helpers/convertUnixTimestampToDate';

export default {
  title: 'CardTemplate/CardTemplate',
  component: CardTemplate,
  argTypes: {
    cardProductName: {
      type: 'string',
      defaultValue: 'Smart Debit card',
      options: ['Credit Card Billable', 'Credit Card Billable Premium', 'Smart Debit card', 'Debit Travel card'],
      control: { type: 'select' },
    },
    paymentSystem: {
      name: 'Visa/MasterCard',
      type: 'string',
      defaultValue: PaymentSystem.MASTER_CARD,
      description: 'Visa or MasterCard type',
      options: [PaymentSystem.MASTER_CARD, PaymentSystem.VISA],
      control: { type: 'radio' },
    },
    cardDateStoryBook: {
      type: 'number',
      description: 'The end date of card',
      defaultValue: 1663707600000,
      control: { type: 'date' },
    },
    cardNumber: {
      type: 'string',
      defaultValue: '7777',
      description: 'The last four digits of the card number',
    },
    currency: {
      type: 'string',
      defaultValue: 'GBP',
      description: 'Currency',
    },
    cvv: {
      type: 'string',
      defaultValue: '***',
      description: 'cvv code',
    },
  },
  parameters: {
    docs: {
      description: {
        component: 'Cards',
      },
    },
    layout: 'centered',
  },
} as ComponentMeta<typeof CardTemplate>;

interface Props extends CardTemplateProps {
  cardDateStoryBook: number;
}

const Template = ({
  id = 1,
  cardProductName,
  paymentSystem,
  cardType,
  cardNumber,
  currency,
  cardDateStoryBook,
  cvv = '111',
}: Props): ReactElement => (
  <CardTemplate
    id={id}
    cardProductName={cardProductName}
    paymentSystem={paymentSystem}
    cardType={cardType}
    cardNumber={cardNumber}
    cardDate={convertUnixTimestampToDate(cardDateStoryBook)}
    currency={currency}
    cvv={cvv}
  />
);

export const Default = Template.bind({});
