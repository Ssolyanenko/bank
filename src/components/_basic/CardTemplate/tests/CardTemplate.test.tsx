import { shallow } from 'enzyme';

import { CardTemplate } from 'components/_basic';
import { CardTemplate as CardTemplateProps } from 'interfaces/cardTemplate';
import { Default } from 'components/_basic/CardTemplate/CardTemplate.stories';

describe('<CardTemplate />', (): void => {
  const props: CardTemplateProps = {
    id: 1,
    cardNumber: '1',
    cardProductName: 'Debit Travel card',
    formattedAmount: '1000',
    currency: 'GBP',
    cardDate: '12/24',
    paymentSystem: 'visa',
  };

  test('should make a snapshot of component', (): void => {
    const wrapper = shallow(<CardTemplate {...props} />);

    expect(wrapper.getElement()).toMatchSnapshot();
  });

  test('should render CardTemplate from storybook', (): void => {
    const wrapper = shallow(<Default {...props} cardDateStoryBook={1663707600000} />);

    expect(wrapper.find(CardTemplate).length).toBe(1);
  });
});
