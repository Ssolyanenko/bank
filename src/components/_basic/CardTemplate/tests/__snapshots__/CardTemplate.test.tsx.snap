// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardTemplate /> should make a snapshot of component 1`] = `
<div
  className="cardWrappingArea "
>
  <div
    className="cardWrapper "
    onClick={[Function]}
    role="presentation"
  >
    <main
      className=" card debitTravel  "
    >
      <div
        className="nameWrapper"
      >
        <h3
          className="name"
        >
          Travel card
        </h3>
        <AppLogoIcon
          className="logoIcon"
          color="white_1"
        />
      </div>
      <div
        className="creditNumber"
      >
        <CardTemplateDots />
        <CardTemplateDots />
        <CardTemplateDots />
        <div
          className="dotNumbers"
        >
          1
        </div>
      </div>
      <div
        className="cardDate"
      >
        12/24
      </div>
      <div
        className="totalSumWrapper"
      >
        <p
          className="totalSum"
        >
          1000 GBP
        </p>
        Unknown type of card
      </div>
    </main>
    <div
      className=" card debitTravel cardBack"
    >
      <div
        className="stripe"
      />
      <div
        className="cvv"
      >
        <div
          className="showPassword"
        >
          <React.Fragment>
            <span
              className="stars"
            >
               *** 
            </span>
            <span
              className="icon"
            >
              <PasswordIconInvisible
                onClick={[Function]}
              />
            </span>
          </React.Fragment>
        </div>
      </div>
    </div>
  </div>
</div>
`;
